url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/spec/args/ArgListBuilderUtil.java

```
@@ -0,0 +1,714 @@
+
+        @ScriptyCommand(name="InstanceOrBinding")
+        public static TypeTuple instanceOrBindingType(IContext aCtx, Object[] aArgs)
+        throws CommandException
+        {
+            try
+            {
+                aArgs = INSTANCEORBINDING_ARGLIST.guard(aArgs, aCtx);
+
+                TypeTuple lTuple = (TypeTuple) aArgs[1];
+                Object lTypeCandidate = lTuple.getX();
+                Map<String, IArgMapping> lMappings = lTuple.getY();
+
+                if(lTypeCandidate == null || !(lTypeCandidate instanceof ITypeSpec))
+                {
+                    throw new CommandException("Invalid type expression InstanceOrBinding(???) for the instance type.");
+                }
+
+                ITypeSpec lSpec = (ITypeSpec) lTypeCandidate;
+
+                return new TypeTuple(new InstanceOrBinding(lSpec), lMappings);
+            }
+            catch (ArgSpecException e)
+            {
+                throw new CommandException(e.getMessage());
+            }
+        }
+
+        // type="Instance com.sdicons.MyClass nullAllowed=true"
+        private static StdArgList INSTANCE_ARGLIST = new StdArgList(new FixedArg[]{FIXED_CLASSNAME}, new OptionalArg[]{}, new NamedArg[]{NAMED_NULLALLOWED});
+
+        @ScriptyCommand(name="Instance")
+        public static TypeTuple instanceType(IContext aCtx, Object[] aArgs)
+        throws CommandException
+        {
+            try
+            {
+                aArgs = INSTANCE_ARGLIST.guard(aArgs, aCtx);
+                String lClassName = (String) aArgs[1];
+                boolean lNullAllowed = (Boolean) aArgs[2];
+
+                try
+                {
+                    Class lClass = Class.forName(lClassName);
+                    return new TypeTuple(new InstanceType(lClass, lNullAllowed), null);                    
+                } 
+                catch (ClassNotFoundException e)
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 265.